pkg/runtime/processor/processor.go

```diff
@@ -179,6 +183,11 @@ func New(opts Options) *Processor {
 		Meta:           opts.Meta,
 	})
 
+	// ensure a default no-op reporter
+	if opts.Reporter == nil {
+		opts.Reporter = DefaultReporter
+	}
```

Comment:
Please don't write to options.

```suggestion
	reporter := DefaultReporter
	if opts.Reporter != nil {
	  reporter = opts.Reporter
	}
```